### PR TITLE
refactor: plugin manifest utils

### PR DESCRIPTION
**Pull request by yuqizhou77**
- util functions:
  -  get plugin file path from Teams manifest
  -  get api spec from Teams manifest or plugin manifest
- list existing APIs from the referenced api spec if it is plugin. If API ME, we still check the commands in the manifest, and list operations with operation id matching command id.
- Update listOperations and CopilotAddApi to list operations based on capability
- Update strings

### CODE REVIEW

**Review by nliu-ms — DISMISSED** [review]

**Review by jayzhang — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -287,8 +289,8 @@
   "core.createProjectQuestion.projectType.outlookAddin.title": "App Features Using an Outlook Add-in",
   "core.createProjectQuestion.projectType.tab.detail": "Embed your own web content in Teams, Outlook, and the Microsoft 365 app",
   "core.createProjectQuestion.projectType.tab.title": "App Features Using a Tab",
-  "core.createProjectQuestion.projectType.copilotPlugin.detail": "Create a plugin to extend Microsoft 365 Copilot using your APIs",
-  "core.createProjectQuestion.projectType.copilotPlugin.label": "Plugin",
+  "core.createProjectQuestion.projectType.copilotPlugin.detail": "Create a plugin to extend Microsoft Copilot for Microsoft 365 using your APIs",
+  "core.createProjectQuestion.projectType.copilotPlugin.label": "Plugin for Copilot",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:293]: Suggestion: Copilot Plugin
Logic: Easier to perceive

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -287,8 +289,8 @@
   "core.createProjectQuestion.projectType.outlookAddin.title": "App Features Using an Outlook Add-in",
   "core.createProjectQuestion.projectType.tab.detail": "Embed your own web content in Teams, Outlook, and the Microsoft 365 app",
   "core.createProjectQuestion.projectType.tab.title": "App Features Using a Tab",
-  "core.createProjectQuestion.projectType.copilotPlugin.detail": "Create a plugin to extend Microsoft 365 Copilot using your APIs",
-  "core.createProjectQuestion.projectType.copilotPlugin.label": "Plugin",
+  "core.createProjectQuestion.projectType.copilotPlugin.detail": "Create a plugin to extend Microsoft Copilot for Microsoft 365 using your APIs",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:292]: Suggestion: Create a plugin to extend Copilot for Microsoft 365 using your APis.
Logic: 'Microsoft' is redundant if used twice and user would anyway know that it is the Microsoft Copilot.

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:293]: @MuyangAmigo what's your opinion on this? I thought we have Figma designed as "Copilot plugin". Should we update as what Surpiya suggested and update the Figma?

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:292]: @MuyangAmigo same question about this string.

Comment (MuyangAmigo) [reply packages/fx-core/resource/package.nls.json:292]: Hi @supkasar - `Microsoft Copilot for Microsoft 365` is a fixed branding term. https://learn.microsoft.com/en-us/microsoft-365-copilot/microsoft-365-copilot-overview

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (MuyangAmigo) [reply packages/fx-core/resource/package.nls.json:293]: OK we could chang to Copilot Plugin.

**Review by MuyangAmigo — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json]: @supkasar updated as suggested.

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (supkasar) [reply packages/fx-core/resource/package.nls.json:292]: Ok, thanks for clarification. In that case, no change required.

Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:292]: Would you mind sign off the string part of this PR? @supkasar

Comment (supkasar) [reply packages/fx-core/resource/package.nls.json:292]: Sure, @yuqizhou77 will do that.

**Review by supkasar — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**